src/libraries/System.Private.Xml.Linq/tests/misc/XAttribute.cs

```diff
@@ -17,5 +16,73 @@ public void FormattedDate()
             // Ensure we are compatible with the .NET Framework
             Assert.Equal("CreatedTime=\"2018-01-01T12:13:14Z\"", new XAttribute("CreatedTime", new DateTime(2018, 1, 1, 12, 13, 14, DateTimeKind.Utc)).ToString());
         }
+
+        public static IEnumerable<object[]> NumericValuesWithMinusSign()
+        {
+            yield return new object[] { -123 };
+            yield return new object[] { -123f };
+            yield return new object[] { -123L };
+            yield return new object[] { (short)-123 };
+            yield return new object[] { -12.3 };
+            yield return new object[] { -12.3m };
+            yield return new object[] { (sbyte)-123 };
+        }
+
+        [Theory]
+        [MemberData(nameof(NumericValuesWithMinusSign))]
+        public void MinusSignWithDifferentTypeSwedishCulture(object value)
+        {
+            CultureInfo newCulture = null;
+            try
+            {
+                newCulture = new CultureInfo("SE-SW");
+            }
+            catch (CultureNotFoundException) { /* Do nothing */ }
+
+            using (new ThreadCultureChange(newCulture))
+            {
+
+                Assert.Equal('-', (new XAttribute("a", value)).Value[0]);
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(NumericValuesWithMinusSign))]
+        public void MinusSignWithDifferentTypeNoCulture(object value)
+        {
+            Assert.Equal('-', (new XAttribute("a", value)).Value[0]);
+        }
+
+        public static IEnumerable<object[]> NonNumericValues()
+        {
+            yield return new object[] { true, "true" };
+            yield return new object[] { new DateTimeOffset(2018, 1, 1, 12, 13, 14, TimeSpan.Zero), "2018-01-01T12:13:14Z" };
+            yield return new object[] { new TimeSpan(12, 13, 14), "PT12H13M14S" };
+            yield return new object[] { "-123\n", "-123\n" };
+        }
+
+        [Theory]
+        [MemberData(nameof(NonNumericValues))]
+        public void NonNumericTypeSwedishCulture(object value, string expected)
+        {
+            CultureInfo newCulture = null;
+            try
+            {
+                newCulture = new CultureInfo("SE-SW");
+            }
+            catch (CultureNotFoundException) { /* Do nothing */ }
```

Comment:
should it return in the catch block? ie on a machine where for some reason does not have this culture, you want it to silently continue and use the current culture instead?

Reply:
Yeah, i want it continue even the culture not found, because i don't see much test in this class, so still is not useless, plus the issue could happen for other culture too, wherever the test run its worth testing it